mws/response.py

```
@@ -0,0 +1,201 @@
+        "metadata",
+        "request_id",
+    ]
+
+    def __init__(self, response, result_key=None, encoding=None, force_cdata=False):
+        super().__init__(response)
+        self.timestamp = None
+        self._result_key = result_key
+
+        if not self.encoding:
+            # If the response did not specify its encoding,
+            # we use either A) an encoding specified by the user,
+            # or B) the ``apparent_encoding`` of the ``requests.Response`` object,
+            # which uses ``chardet`` to guess the encoding of the response content.
+            # Either way, we need an encoding saved in order to parse the content
+            # from XML into DotDicts.
+            self.encoding = encoding or response.apparent_encoding
+
+        self._dict = None
+        self._dotdict = None
+        self._metadata = None
+        self.parse_response(force_cdata=force_cdata)
+
+    def __repr__(self):
+        return "<{} [{}]>".format(self.__class__.__name__, self.original.status_code)
+
+    def parse_response(self, force_cdata=False):
+        """Runs :py:meth:`.text <.text>` through ``xmltodict.parse()``, storing the
+        returned Python dictionary as ``._dict``.
+
+        If no XML errors occur during that process, constructs
+        :py:class:`DotDict <mws.collections.DotDict>` instances
+        from the parsed XML data, making them available from
+        :py:meth:`.parsed <.parsed>` and :py:meth:`.metadata <.metadata>`.
+
+        For non-XML responses, does nothing.
+
+        :param bool force_cdata: Passed to ``xml_to_dict.parse()`` when
+         parsing XML content. Defaults to ``False``. Ignored for non-XML responses.
+        """
+        try:
+            # Attempt to convert text content to an
+            self._dict = mws_xml_to_dict(
+                self.content, encoding=self.encoding, force_cdata=force_cdata
+            )
+        except ExpatError:
+            # Probably not XML content: just ignore it.
```

Dangerous?

---

Taken in context:

```python
try:
    # Attempt to convert text content to an
    self._dict = mws_xml_to_dict(
        self.content, encoding=self.encoding, force_cdata=force_cdata
    )
except ExpatError:
    # Probably not XML content: just ignore it.
    pass
else:
    # No exception? Cool
    self._build_dotdicts()
```

The `ExpatError` is likely to occur for a response that's not actually XML, and the method doesn't really have to do much else from there.

`else` in a `try` block only executes if no exception occurs, even if the exception was ignored. So, if the error occurs, no other action is taken.